### ISSUE TITLE
Doc:  update how to use official image or build image

### ISSUE THREAD

**Issue by jichenjc** [kubernetes-sigs/cluster-api-provider-openstack#463]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

@sbueringer sorry, I encountered following issue in VM start, seems no kubeadm installed at all, and I checked the kubeadmconfig file generated, seems no kubeadm /kubectl install
this belongs to CABPK or CAPO? should we update the generate file to add this? 

**What did you expect to happen:**
Sep  9 02:35:02 test-cluster-controlplane-0 cloud-init[1315]: /var/lib/cloud/instance/scripts/runcmd: 2: /var/lib/cloud/instance/scripts/runcmd: kubeadm: not found
Sep  9 02:35:02 test-cluster-controlplane-0 cloud-init[1315]: Cloud-init v. 19.1-1-gbaa47854-0ubuntu1~16.04.1 running 'modules:final' at Mon, 09 Sep 2019 02:35:01 +0000. Up 53.23 seconds.
Sep  9 02:35:02 test-cluster-controlplane-0 cloud-init[1315]: 2019-09-09 02:35:02,017 - util.py[WARNING]: Failed running /var/lib/cloud/instance/scripts/runcmd [127]
Sep  9 02:35:02 test-cluster-controlplane

my cloud-init script:
```
-   path: /tmp/kubeadm.yaml
    owner: root:root
    permissions: '0640'
    content: |
      ---
      ---
      apiServer:
        extraArgs:
          cloud-config: /etc/kubernetes/cloud.conf
          cloud-provider: openstack
        extraVolumes:
        - hostPath: /etc/kubernetes/cloud.conf
          mountPath: /etc/kubernetes/cloud.conf
          name: cloud
          readOnly: true
      apiVersion: kubeadm.k8s.io/v1beta1
      certificatesDir: ""
      clusterName: test-cluster
      controlPlaneEndpoint: 9.20.205.29:6443
      controllerManager:
        extraArgs:
          cloud-config: /etc/kubernetes/cloud.conf
          cloud-provider: openstack
        extraVolumes:
        - hostPath: /etc/kubernetes/cloud.conf
          mountPath: /etc/kubernetes/cloud.conf
          name: cloud
          readOnly: true
        - hostPath: /etc/certs/cacert
          mountPath: /etc/certs/cacert
          name: cacerts
          readOnly: true
      dns:
        type: ""
      etcd: {}
      imageRepository: k8s.gcr.io
      kind: ClusterConfiguration
      kubernetesVersion: 1.15.0
      networking:
        dnsDomain: cluster.local
        podSubnet: 192.168.0.0/16
        serviceSubnet: 10.96.0.0/12

      ---
      apiVersion: kubeadm.k8s.io/v1beta1
      kind: InitConfiguration
      localAPIEndpoint:
        advertiseAddress: '{{ ds.ec2_metadata.local_ipv4 }}'
        bindPort: 6443
      nodeRegistration:
        kubeletExtraArgs:
          cloud-config: /etc/kubernetes/cloud.conf
          cloud-provider: openstack
        name: '{{ local_hostname }}'

runcmd:
  - 'kubeadm init --config /tmp/kubeadm.yaml'
ntp:
  servers:
users:
  - name: cloudusr
    ssh_authorized_keys:
      - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDGPRj84C8TWrvr8qS2ZL7gysKp56bXP5EjSgG2Bxh89p6S/V+6Ix8NnkybA5Sl21KISZjd9PPtBqen4L/pQRpqqdGWCWPnViNFkNMDYXnkAD1l12Ft57auBFbTv9Ru4iXQlIW80NwHnIlDTIemhOrciK3iXuJ70fheXK9gQCaRMm8cLL3XaMkCQOqJgkH6auOMSZhYMDas7ZoDt/dbBufwZ6bZetv1WBgDa9JHNxQeC0/y3ageUVye1kLCowAsNZVf/CxwJudOc9Tn6ja9oFbyJRuG193JNhPpA/G27xitd5+a+eMaoxFCgWNmfpQehGYSGtAACb/kBsqqxh8FJwyH

```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by jichenjc** [comment]
so I am still catchup with the new model, anything related to CAPKB is helpful to me.. thanks

**Comment by sbueringer** [comment]
Of course no problem :). The other providers now rely on images build like this: https://github.com/kubernetes-sigs/image-builder/tree/master/images/capi (at lesat CAPA & CAPV). In these images stuff like container/kubeadm/kubectl/crictl are already pre-installed (which makes sense because not every machine has to install them by itself on startup), e.g.: https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/ansible/roles/kubernetes/tasks/debian.yml#L29

I think there are two options:
* Adjust the images used so all necessary software is already instaled
* Install the images via scripts in KubeadmConfig (configure shell script and execute before kubeadm: https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/blob/master/api/v1alpha2/kubeadmbootstrapconfig_types.go#L47-L50

So basically all is possible. If possible, I would just like to document the way with the official images and not maintain additional install scripts for this stuff. But might make sense to document this

**Comment by jichenjc** [comment]
yes, this approach looks good to me , I will modify the title of this issue to a doc change request

**Comment by randomvariable** [comment]
If at all possible, would be good to converge on using https://github.com/kubernetes-sigs/image-builder

**Comment by sbueringer** [comment]
Exactly my opinion. We should more or less just document that we rely on image-builder/capi images and everything else should be document in their repository.

**Comment by jichenjc** [comment]
ok, this make sense , will update soon

**Comment by hidekazuna** [comment]
I tried to convert published OVA image to qcow2 manually and use it,  but failing.
Does anyone succeed to test v1alpha2 based on KVM?

**Comment by randomvariable** [comment]
@hidekazuna Something along the lines of http://ask.xmodulo.com/convert-ova-to-qcow2-linux.html should work?

**Comment by sbueringer** [comment]
@hidekazuna I think I did something like that. First build the VMDK and use it for our VIO and then convert it to qcow2 and run it on the OpenStack of @chrigl (but not sure if it's using KVM, but I guess it does)

**Comment by jichenjc** [comment]
are you having problem in using the image only this time(after v1alpha2)? or it should be a general problem that ? I think most of the system we are testing is on KVM ..

**Comment by hidekazuna** [comment]
@randomvariable I followed the procedure you proposed. Unzipped [the image](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere#kubernetes-versions-with-published-ovas) and convert vmdk to qcow2. After that I registered it to glance and use it.

@jichenjc I have the problem after v1alpha2. Because we can use kubeadm pre-installed image with v1alpha2.
